lib/core/src/persist/mod.rs

```diff
@@ -228,11 +228,21 @@ impl Persister {
                 return Ok(());
             }
         };
-        let maybe_script_pubkey = tx
+        let maybe_address = tx
             .outputs
             .iter()
             .find(|output| output.is_some())
-            .and_then(|output| output.clone().map(|o| o.script_pubkey.to_hex()));
+            .and_then(|output| {
+                output.clone().and_then(|o| {
+                    o.address
+                        .blinding_pubkey
+                        .ok_or_else(|| anyhow!("Missing blinding pubkey for address"))
+                        .map(|blinding_pubkey| {
+                            o.address.to_confidential(blinding_pubkey).to_string()
+                        })
+                        .ok()
```

Comment:
Could this be simplified?
```suggestion
                    o.address.blinding_pubkey.map(|blinding_pubkey| {
                        o.address.to_confidential(blinding_pubkey).to_string()
                    })
```

Reply:
Fixed! Thanks